// docpad-plugin-uglify/out/uglify.plugin.js
// Generated by CoffeeScript 1.6.3
var __hasProp = {}.hasOwnProperty,
  __extends = function(child, parent) { for (var key in parent) { if (__hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; };

module.exports = function(BasePlugin) {
  var UglifyPlugin, _ref;
  return UglifyPlugin = (function(_super) {
    __extends(UglifyPlugin, _super);

    function UglifyPlugin() {
      _ref = UglifyPlugin.__super__.constructor.apply(this, arguments);
      return _ref;
    }

    UglifyPlugin.prototype.name = 'uglify';

    UglifyPlugin.prototype.config = {
      all: true
    };

    UglifyPlugin.prototype.renderDocument = function(opts) {
      var extension, file;
      file = opts.file, extension = opts.extension;
      if (extension === 'js') {
        if ((this.getConfig().all === true && file.get('minify') !== false) || (file.get('minify') === true)) {
          opts.content = require('uglify-js').minify(opts.content, {
            fromString: true
          }).code;
        }
      }
      return this;
    };

    return UglifyPlugin;

  })(BasePlugin);
};
